mirgecom/euler.py

```diff
@@ -95,44 +106,86 @@ def euler_operator(discr, state, gas_model, boundaries, time=0.0):
         Physical gas model including equation of state, transport,
         and kinetic properties as required by fluid state
 
+    quadrature_tag
+
+        An optional identifier denoting a particular quadrature
+        discretization to use during operator evaluations.
+        The default value is *None*.
+
     Returns
     -------
-    numpy.ndarray
+    :class:`mirgecom.fluid.ConservedVars`
 
         Agglomerated object array of DOF arrays representing the RHS of the Euler
         flow equations.
     """
-    from mirgecom.gas_model import project_fluid_state
-    boundary_states = {btag:
-                       project_fluid_state(discr, "vol", btag, state, gas_model)
-                       for btag in boundaries}
-
-    interior_cv = interior_trace_pairs(discr, state.cv)
-
-    # If this is a mixture, we need to exchange the temperature field because
-    # mixture pressure (used in the inviscid flux calculations) depends on
-    # temperature and we need to seed the temperature calculation for the
-    # (+) part of the partition boundary with the remote temperature data.
-    tseed_interior_pairs = (interior_trace_pairs(discr, state.temperature)
-                            if state.is_mixture else None)
-
-    from mirgecom.gas_model import make_fluid_state_trace_pairs
-    interior_states = make_fluid_state_trace_pairs(interior_cv, gas_model,
-                                                   tseed_interior_pairs)
+    dd_base_vol = DOFDesc("vol")
+    dd_quad_vol = DOFDesc("vol", quadrature_tag)
+    dd_quad_faces = DOFDesc("all_faces", quadrature_tag)
+
+    def _interp_to_surf_quad(utpair):
+        local_dd = utpair.dd
+        local_dd_quad = local_dd.with_discr_tag(quadrature_tag)
+        return TracePair(
+            local_dd_quad,
+            interior=op.project(discr, local_dd, local_dd_quad, utpair.int),
+            exterior=op.project(discr, local_dd, local_dd_quad, utpair.ext)
+        )
```

Comment:
Does `op.project(utpair)` work? It should.

Reply:
@inducer It doesn't update `dd`, which is why this is here.

Reply:
OIC. Add a comment?

Reply:
(781b8fc)